v2/tests/log_context_property_type_wchar_t_ptr_int/log_context_property_type_wchar_t_ptr_int.c

```diff
@@ -0,0 +1,254 @@
+    POOR_MANS_ASSERT(wcscmp(dst_buffer, L"cucu") == 0);
+}
+
+/* LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).free */
+
+/* Tests_SRS_LOG_CONTEXT_PROPERTY_TYPE_WCHAR_T_PTR_07_011: [ LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).free shall return. ]*/
+static void wchar_t_ptr_free_returns(void)
+{
+    // arrange
+    wchar_t buffer[128];
+    LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr)(buffer, wcslen(L"cucu") + 1, L"cucu");
+
+    // act
+    LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).free(buffer);
+
+    // assert
+    // no explicit assert, no crash expected
+}
+
+/* LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).get_type */
+
+static void wchar_t_ptr_get_type_returns_LOG_CONTEXT_PROPERTY_TYPE_wchar_t_ptr(void)
+{
+    // arrange
+
+    // act
+    LOG_CONTEXT_PROPERTY_TYPE result = LOG_CONTEXT_PROPERTY_TYPE_IF_IMPL(wchar_t_ptr).get_type();
+
+    // assert
+    POOR_MANS_ASSERT(result == LOG_CONTEXT_PROPERTY_TYPE_wchar_t_ptr);
+}
+
+/* LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr) */
+
+static void wchar_t_ptr_LOG_CONTEXT_PROPERTY_TYPE_INIT_with_NULL_dst_value_fails(void)
+{
+    // arrange
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr)(NULL, wcslen(L"cucu") + 1, L"cucu");
+
+    // assert
+    POOR_MANS_ASSERT(result != 0);
+}
+
+static void wchar_t_ptr_LOG_CONTEXT_PROPERTY_TYPE_INIT_with_NULL_format_fails(void)
+{
+    // arrange
+    wchar_t buffer[5];
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr)(buffer, 1, NULL);
+
+    // assert
+    POOR_MANS_ASSERT(result != 0);
+}
+
+static void wchar_t_ptr_LOG_CONTEXT_PROPERTY_TYPE_INIT_succeeds(void)
+{
+    // arrange
+    wchar_t buffer[5];
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr)(buffer, wcslen(L"cucu") + 1, L"cucu");
+
+    // assert
+    POOR_MANS_ASSERT(result == 0);
+    POOR_MANS_ASSERT(wcscmp(buffer, L"cucu") == 0);
+}
+
+static void wchar_t_ptr_LOG_CONTEXT_PROPERTY_TYPE_INIT_with_multiple_args_succeeds(void)
+{
+    // arrange
+    wchar_t buffer[43];
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_INIT(wchar_t_ptr)(buffer, 50, L"The answer is %d and let's say hello %ls", 42, L"world");
+
+    // assert
+    POOR_MANS_ASSERT(result == 0);
+    POOR_MANS_ASSERT(wcscmp(buffer, L"The answer is 42 and let's say hello world") == 0);
+}
+
+/* LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(wchar_t_ptr)*/
+
+static void wchar_t_ptr_LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE_succeeds(void)
+{
+    // arrange
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(wchar_t_ptr)(L"cucu");
+
+    // assert
+    POOR_MANS_ASSERT(result == 5);
+}
+
+static void wchar_t_ptr_LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE_with_multiple_args_succeeds(void)
+{
+    // arrange
+
+    // act
+    int result = LOG_CONTEXT_PROPERTY_TYPE_GET_INIT_DATA_SIZE(wchar_t_ptr)(L"The answer is %d and let's say hello %ls", 43, L"world");
+
+    // assert
+    POOR_MANS_ASSERT(result == 43);
```

Comment:
>43 [](http://example.com/codeflow?start=31&length=2)

why is this 43 and not 39?

Reply:
I've counted (with chatbot) as 39 the number of characters...

Reply:
size is 43 units of wide char, changed to 43*sizeof(wchar_t)